torch_geometric/_onnx.py

```diff
@@ -12,3 +16,213 @@ def is_in_onnx_export() -> bool:
+        >>>
+        >>> model = MyModel()
+        >>> x = torch.randn(3, 8)
+        >>> edge_index = torch.tensor([[0, 1, 2], [1, 0, 2]])
+        >>> success = safe_onnx_export(model, (x, edge_index), 'model.onnx')
+        >>>
+        >>> # For CI environments:
+        >>> success = safe_onnx_export(model, (x, edge_index), 'model.onnx',
+        ...                             skip_on_error=True)
+        >>> if not success:
+        ...     print("ONNX export skipped due to known upstream issue")
+    """
+    # Convert single tensor to tuple for torch.onnx.export compatibility
+    if isinstance(args, torch.Tensor):
+        args = (args, )
+
+    try:
+        # First attempt: standard ONNX export
+        torch.onnx.export(model, args, f, **kwargs)
+        return True
+
+    except Exception as e:
+        error_str = str(e)
+        error_type = type(e).__name__
+
+        # Check for the specific onnx_ir.serde.SerdeError patterns
+        is_allowzero_error = (('onnx_ir.serde.SerdeError' in error_str
+                               and 'allowzero' in error_str) or
+                              'ValueError: Value out of range: 1' in error_str
+                              or 'serialize_model_into' in error_str
+                              or 'serialize_attribute_into' in error_str)
+
+        if is_allowzero_error:
+            warnings.warn(
+                f"Encountered known ONNX serialization issue ({error_type}). "
+                "This is likely the allowzero boolean attribute bug. "
+                "Attempting workaround...", UserWarning, stacklevel=2)
+
+            # Apply workaround strategies
+            return _apply_onnx_allowzero_workaround(model, args, f,
+                                                    skip_on_error, **kwargs)
+
+        else:
+            # Re-raise other errors
+            raise
+
+
+def _apply_onnx_allowzero_workaround(
+    model: torch.nn.Module,
+    args: tuple[Any, ...],
+    f: Union[str, PathLike[Any], None],
+    skip_on_error: bool = False,
+    **kwargs: Any,
+) -> bool:
+    r"""Apply workaround strategies for onnx_ir.serde.SerdeError with allowzero
+    attributes.
+
+    Returns:
+        bool: True if export succeeded, False if skipped (when
+              skip_on_error=True).
+    """
+    # Strategy 1: Try without dynamo if it was enabled
+    if kwargs.get('dynamo', False):
+        try:
+            kwargs_no_dynamo = kwargs.copy()
+            kwargs_no_dynamo['dynamo'] = False
+
+            warnings.warn(
+                "Retrying ONNX export with dynamo=False as workaround",
+                UserWarning, stacklevel=3)
+
+            torch.onnx.export(model, args, f, **kwargs_no_dynamo)
+            return True
+
+        except Exception:
+            pass
+
+    # Strategy 2: Try with different opset versions
+    original_opset = kwargs.get('opset_version', 18)
+    for opset_version in [17, 16, 15, 14, 13, 11]:
```

Comment:
It doesn't seem to be the best idea to try different opsets to get around an onnx-ir bug. It would be helpful to advise users to update the package instead so they can stay on a newer opset. Older opsets (<18) prevents important optimizations possible on the model and are not recommended.